src/AbstractWalkers.jl

```diff
@@ -0,0 +1,147 @@
+"""
+Module containing abstract definitions for walkers.
+"""
+module AbstractWalkers
+
+using AtomsBase
+using Unitful
+using ..Potentials
+using ..EnergyEval
+
+export AtomWalker, AtomWalkers, LJAtomWalkers
+export update_walker!
+
+abstract type AtomWalkers end
+
+"""
+    mutable struct AtomWalker
+
+The `AtomWalker` struct represents a walker composed of atoms/molecules.
+
+# Fields
+- `configuration::FastSystem`: The configuration of the walker.
+- `energy::typeof(0.0u"eV")`: The energy of the walker.
+- `iter::Int64`: The current iteration number of the walker.
+- `num_frozen_part::Int64`: The number of frozen particles in the walker.
+- `energy_frozen_part::typeof(0.0u"eV")`: The energy of the frozen particles in the walker, serves as a constant energy offset
+    to the interacting part of the system.
+
+# Constructor
+```julia
+AtomWalker(configuration::FastSystem; energy=0.0u"eV", iter=0, num_frozen_part=0, energy_frozen_part=0.0u"eV")
+```
+Create a new `AtomWalker` with the given configuration and optional energy, iteration number, number of frozen particles, and energy of the frozen particles.
+
+"""
+mutable struct AtomWalker 
+    configuration::FastSystem
+    energy::typeof(0.0u"eV")
+    iter::Int64
+    num_frozen_part::Int64
+    energy_frozen_part::typeof(0.0u"eV")
+    function AtomWalker(configuration::FastSystem; energy=0.0u"eV", iter=0, num_frozen_part=0, energy_frozen_part=0.0u"eV")
+        return new(configuration, energy, iter, num_frozen_part, energy_frozen_part)
+    end
+end
+
+"""
+    assign_lj_energies!(walkers::Vector{AtomWalker}, lj::LJParameters)
+
+Assigns LJ energies to a list of AtomWalkers.
+
+# Arguments
+- `walkers::Vector{AtomWalker}`: A list of AtomWalkers to assign energies to.
+- `lj::LJParameters`: The LJ parameters used for energy calculation.
+
+# Returns
+- `walkers::Vector{AtomWalker}`: The updated list of AtomWalkers with assigned energies.
+"""
+function assign_lj_energies!(walkers::Vector{AtomWalker}, lj::LJParameters)#; frozen::Int64=0, e_frozen=0.0u"eV")
+    for walker in walkers
+        if walker.num_frozen_part > 0
```

Comment:
`assign_lj_energies` is only evaluated at the beginning of the run, correct?

Reply:
Yes and no, it assigns the LJ energy whenever a `LJAtomWalkers` object is constructed. `LJAtomWalkers` has two fields, `walkers` and `lj_potential`, the idea here is whenever these two fields are combined, the energy of the `walkers` should update according to the `lj_potential`. For example, if you restart a job with a different set of LJ parameters, the energies of walkers from a snapshot that were previously saved are no longer correct hence should be updated. Of course, at the beginning of the run this function also assigns energies to the walkers which don't yet have a starting energy.